packages/asu-core/src/aws/lambda/error-handler.ts

```diff
@@ -1,25 +1,28 @@
 import { APIGatewayProxyResult } from "aws-lambda";
-import { DataValidationError } from "../../asu-core";
-import { NotFoundError } from "../../errors/not-found.error";
+import {DataValidationError, NotFoundError, UnauthenticatedError} from "../../asu-core";
 
 export const lambdaHandleError = (error: any): APIGatewayProxyResult => {
-  if (error instanceof NotFoundError) {
-    return {
-      statusCode: 404,
-      body: error.message
-    }
-  }
-  else if (error instanceof DataValidationError) {
-    return {
-      statusCode: 400,
-      body: error.message
-    }
-  }
-  else {
-    console.log(`lambdaHandleError: Unhandled exception: ${error}`);
-    return {
-      statusCode: 500,
-      body: 'Internal Service Error'
-    }
+  switch(error.constructor) {
+    case NotFoundError:
+      return {
+        statusCode: 404,
+        body: error.message
+      }
+    case DataValidationError:
+      return {
+        statusCode: 400,
+        body: error.message
+      }
+    case UnauthenticatedError:
+      return {
+        statusCode: 401,
+        body: error.message
+      }
```

Comment:
Part 2 of the auth request error flow: If we don't want to automatically retry on `authenticationFailure` responses and do just stick with throwing an Unauthenticated error in that case, would we ever want to show the user that 401?

I feel like it could help to give context as to what issue their experiencing with the service, but similarly to our other error handling conversations, it's not like they can change anything about the authentication process to try again.

Reply:
I believe at this point they could change something. There would be something wrong with the way that they are accessing the application at this point.  This is different from our app having the wrong Adobe Sign credentials, this has to do with they themselves authenticating, and they are not authenticated. This is good (: